demos/plot_different_mesh_adaptation.py

```diff
@@ -21,12 +21,14 @@
 plot_dir = create_directory(f"{demo}/plots")
 
 runs = ["uniform", "hessian", "go"]
+linestyle = ["solid", "dashed", "dotted"]
 fig, axes = plt.subplots()
-for run in runs:
+for i, run in enumerate(runs):
     J = np.load(f"{demo}/data/{run}_progress_J_{n}_{method}.npy")
     t = np.load(f"{demo}/data/{run}_progress_t_{n}_{method}.npy")
+    nc_final = np.load(f"{demo}/data/{run}_progress_nc_{n}_{method}.npy")[-1]
     times = [sum(t[:i]) for i in range(len(t))]
-    axes.loglog(times, J, label=run)
+    axes.loglog(times, J, label=f"{run} with {nc_final} elements in final mesh", linestyle=linestyle[i])
```

Comment:
This is still too long for a label. Why not just `f"{run} ({nc_final} elements)"`, with the caveat that this means the final number?